### PR TITLE
more shape colors

### PR DESCRIPTION
**Pull request by flugsio**
adds keybinds for drawing with secondary color for shift-RMB, previous shortcuts shift-LMB and RMB still produce green shapes. Also 2 more colors when adding in Alt key. This doesn't provide color customization, only input method to easily draw different colors.

### CODE REVIEW

File: src/draw.js [hunk 1]
```diff
@@ -64,24 +67,26 @@ function not(f) {
 }
 
 function addCircle(drawable, key) {
+  var drawColor = drawable.current.drawColor;
   var sameCircle = function(s) {
-    return s.length === 1 && s[0] === key;
+    return s.drawColor === drawColor && s.orig === key && !s.dest;
```
Comment (flugsio) [inline src/draw.js]: Supports drawing all 4 different circle colors on the same square inside each other, in contrast to arrows that only allow 1 color.

**Comment by Calinou** [comment]
Some window managers catch Alt.

**Comment by flugsio** [comment]
The extra colors are not really needed, 2 is enough. Those that use Alt instead of the Win key for window management are probably accustomed to missing functionality which use Alt+mouse.

File: src/draw.js [hunk 2]
```diff
@@ -25,7 +25,11 @@ function start(data, e) {
 function processDraw(data) {
   util.requestAnimationFrame(function() {
     var cur = data.drawable.current;
-    if (cur.orig) cur.over = board.getKeyAtDomPos(data, cur.epos, cur.bounds);
+    if (cur.orig) {
+      dest = board.getKeyAtDomPos(data, cur.epos, cur.bounds);
```
Comment (ornicar) [inline src/draw.js]: let's not forget the `var` keyword to avoid creating global variables

**Pull request merged**